components/raftstore/src/store/worker/pd.rs

```diff
@@ -1023,20 +1025,30 @@ where
                 String::from("batch_split"),
             ),
             Task::AutoSplit { split_infos } => {
-                for split_info in split_infos {
-                    if let Ok(Some(region)) =
-                        block_on(self.pd_client.get_region_by_id(split_info.region_id))
-                    {
-                        self.handle_ask_batch_split(
-                            region,
-                            vec![split_info.split_key],
-                            split_info.peer,
-                            true,
-                            Callback::None,
-                            String::from("auto_split"),
-                        );
+                let pd_client = self.pd_client.clone();
+                let router = self.router.clone();
+                let scheduler = self.scheduler.clone();
+
+                let f = async move {
+                    for split_info in split_infos {
+                        if let Ok(Some(region)) =
+                            pd_client.get_region_by_id(split_info.region_id).await
+                        {
+                            Self::handle_ask_batch_split(
+                                router.clone(),
+                                scheduler.clone(),
+                                pd_client.clone(),
+                                region,
+                                vec![split_info.split_key],
+                                split_info.peer,
+                                true,
+                                Callback::None,
+                                String::from("auto_split"),
+                            );
+                        }
                     }
-                }
+                };
+                spawn_local(f);
```

Comment:
So will it be executed immediately? I think it introduces race between tasks if it's executed later.

Reply:
I think it behaves the same as before because the `handle_ask_batch_split` also needs to wait for the response from PD.

Reply:
I don't get it. How is it related to make `get_region_by_id` async?

Reply:
@BusyJay Can you describe an example which race exists between tasks for this PR?

Reply:
For example, `Task::AutoSplit` and `Task::AskSplit` can arrive almost the same time. In the past, they will be executed one by one. Now auto split can be delay and `AskSplit` is handled first. It's just one of the situations that changes, and perhaps no harm. No sure if there are any other cases that can cause problems.

Reply:
Yes, spawn means we don't care about the sequence. Maybe the original author can end this issue /cc @lhy1024

Reply:
i think we may not care about the change of sequence